packages/common/src/ArcGISContextManager.ts

```diff
@@ -628,6 +636,57 @@ function getTrustedOrgIds(trustedOrgs: IHubTrustedOrgsResponse[]): string[] {
   return trustedOrgs.map((org) => org.to.orgId);
 }
 
+/**
+ * @internal
+ * Get the portal and its limits.
+ * Note: we are only fetching a set of limits,
+ * but we can add additional limits as needed
+ *
+ * @param authentication
+ * @returns {IPortal}
+ */
+async function getSelfWithLimits(
+  authentication: UserSession
+): Promise<IPortal> {
+  const portal = await getSelf({ authentication });
+  const limits = await getPortalLimits(portal.id, authentication);
+  portal.limits = limits;
+
+  return portal;
+}
+
+/**
+ * @internal
+ * Get a set of limits for the given org.
+ * Note: we can add additional limits as needed
+ *
+ * @param orgId
+ * @param authentication
+ */
+async function getPortalLimits(
```

Comment:
This feels like a util fn vs something inside the Context Manager - maybe hoist it into it's own file in common/utils?

Reply:
after discussing, we decided to leave this where it is for now. Once we are on rest-js 4.x, we can re-consider hoisting and refactoring the portal limit utils